rfcs/45-stageless.md

```diff
@@ -0,0 +1,564 @@
+However, each system will continue to work hand-in-hand with a small number of closely-related systems, interlocking in a straightforward and comprehensible way.
+
+By configuring our systems with straightforward, local rules, we can allow the scheduler to pick one of the possible valid paths, optimizing as it sees fit without breaking our logic.
+Just as importantly, we are not over-constraining our ordering, allowing subtle ordering bugs to surface quickly and then be stamped out with a well-considered rule.
+
+### The `App` stores multiple schedules
+
+The `App` can store multiple `Schedules` in a `HashMap<Box<dyn ScheduleLabel>, Schedule>` storage.
+This is used for the enter and exit schedules of states, but can also be used to store, mutate and access additional schedules.
+You can even mutate the schedule that you are in, although you cannot mutate systems that are currently running (failing with an error), as that would be UB.
+
+The main and startup schedules can be accessed using the `CoreSchedule::Main` and `CoreSchedule::Startup` labels respectively.
+By default systems are added to the main schedule.
+You can control this by adding the `.to_schedule(MySchedule::Variant)` system descriptor to your system.
+
+You can access the schedules stored in the app using the `&Schedules` or `&mut Schedules` system parameters.
+Unsurprisingly, these never conflict with entities or resources in the `World`, as they are stored one level higher.
+
+#### Startup systems
+
+Startup systems are stored in their own schedule, with the `CoreSchedule::Startup` label.
+When using the runner added by `MinimalPlugins` and `DefaultPlugins`, this schedule will run exactly once on app startup.
+
+You can add startup systems with the `.add_startup_system(my_system)` method on `App`, which is simply sugar for `.add_system(my_system.to_schedule(CoreSchedule::Startup))`.
+Startup systems can be run again by running this schedule from within an exclusive system.
+
+### Introduction to system configuration
+
+When a function is added to a schedule, it is turned into a `RawSystem` and combined with a `SystemConfig` struct to create a `ConfiguredSystem`.
+
+A system may be configured in the following ways:
+
+- it may have any number of **labels** (which implement the `SystemLabel` trait)
+  - labels are used by other systems to determine ordering constraints
+  - e.g. `.add_system(gravity.label(GameLogic::Physics))`
+  - labels themselves may be configured, allowing you to define high-level structure in your `App` in a single place
+    - e.g `app.configure_label(GameLogic::Physics, SystemConfig::new().label(GameLogic::Physics).after(GameLogic::Input))`
+- it may have ordering constraints, causing it to run before or after other systems
+  - there are several variations on this, see the next section for details
+  - e.g. `.add_system(player_controls.before(GameLogic::Physics))`

```

Comment:
To be consistent with the above example, we might want to find a different example; this (kind of) suggests that `player_controls` is unrelated to `GameLogic::Input` (as it can happen with no regard to when that 'section' happens); I assume that in reality this should be labelled `GameLogic::Input`?